google/cloud/aiplatform/experimental/vertex_model/base.py

```diff
@@ -323,23 +343,24 @@ def p(*args, **kwargs):
 
         # Make remote predictions, regardless of training: create custom container
         if method.__self__.remote:
-            # Convert the predict input to a JSON input for the Endpoint resource
+            # Convert the predict input to a predict_payload input for the Endpoint resource
             data = []
             bound_args = inspect.signature(method).bind(*args, **kwargs)
 
             for parameter_name, parameter in bound_args.arguments.items():
                 if parameter_name == "data":
-                    data = parameter.tolist()
+                    data = obj.predict_input_to_predict_payload(parameter)
                     break
 
             # TODO: cleanup model resource after endpoint is created
             if obj._endpoint is None:
                 _LOGGER.info(
                     "Model is not deployed for remote prediction. Deploying model to an endpoint."
                 )
-                obj._endpoint = obj._model.deploy(machine_type="n1-standard-4")
+                obj._endpoint = obj._model.deploy(machine_type=obj.machine_type)
```

Comment:
Training should also be updated to use this `machine_type`

Reply:
I don't think training uses a machine type -- would it be in the run call?